### DISCUSSION TITLE
How to use Zigbee without any other Server?

### DISCUSSION THREAD

**Discussion by DerPicknicker** [arendst/Tasmota#19222] [Q&A]
Hello all,

I am quite new to Zigbee but not to the esp32 hardware. I use currently an Zigbee stick inside my Homeassistant installation. But now I have an planned installation where I don't want to use an Homeassistant installation - beacause there is no need for. 
Here are my main questions:
- How much Zigbee-Devices can be paired directly to an Tasmota Zigbee Bridge? 
- Can I pair another Device through an already paired device? 
- Can I pair a Tasmota Zigbee gateway to another gateway to extend the range? 
- Can I do some basic rules like: Sensor_1 sends an update and then do a http request? 

Best regards and many thanks for your time!

**Comment by s-hadinger** [comment]
> * How much Zigbee-Devices can be paired directly to an Tasmota Zigbee Bridge?

This depends on the Zigbee MCU. A typical zigbee chip can handle 32 devices as direct children, more via routers. CC2652P (like Sonoff Zigbee Pro) can go hire, maybe ~100.

> * Can I pair another Device through an already paired device?

I'm not sure what you mean. Zigbee has a notion of Router. In practice most mains-powered Zigbee devices (Bulbs, Plugs) also act as router and range extenders. In such case, yes the pairing can be done via the router.

> * Can I pair a Tasmota Zigbee gateway to another gateway to extend the range?

No, you can have only one zigbee coordinator. However you can have routers to act as range extenders (see above)

> * Can I do some basic rules like: Sensor_1 sends an update and then do a http request?

Sure. Doc is here: https://tasmota.github.io/docs/Zigbee/

**Reply by DerPicknicker** [reply]
Hello @s-hadinger ... 

Thank you for the fast response. The main questions are answered. 

The point you didn't get is almost clear for me. I meant that I have the coordinator, and to the coordinator I pared some children (also common named as nodes) .. And I want to pair another children via an children so that the Structure look like this:

Coordinator --> Node 1 (paired to the coordinator) --> Node 2 (paired to Node 1) 

And do you know if there is a real device limit / rule limit in Tasmota? So is it possible for me to have a roule which checks all states of the sensors and send a http request based on the status of the nodes?

**Reply by sfromis** [reply]
As long as "Node 1" is a mains powered Zigbee device, it can normally act as a Zigbee mesh router at the same time as doing the normal operation, meaning that it can relay the connection to/from devices out of reach of the coordinator, but it still needs to be paired with the coordinator first.

Do not expect that you can systematically poll Zigbee sensors, to conserve battery they may not be listening, and only wake up at the interval they are set to transmit.

**Reply by DerPicknicker** [reply]
> 
> 
> 
> Do not expect that you can systematically poll Zigbee sensors, to conserve battery they may not be listening, and only wake up at the interval they are set to transmit.

The "Sensor" is always powered. This should be not the problem. Is there a trigger if Sensor1 sends update do this? I am not very familiar with the Tasmota rule system. And if there is a hard limit on the rules number. Can I simply create a rule which checks 200 sensors and send a specific Http request? 

Thank you!

**Reply by s-hadinger** [reply]
For such specific use cases, you can do this with Berry scripting if your coordinator is based on ESP32.

Are you really planning to have 200 sensors ?

**Reply by DerPicknicker** [reply]
>, you can do this with Berry scripting
> 
> 
> Are you really planning to have 200 sensors ?

Oh great Idea - I thought i need to write 200 Tasmota rules.. the syntax seems like python.. And yes it might happen that I want to have 200 Sensor / devices but not today. It's just for me to choose the right hardware to have no "hard" limits. Or limits which I can prevent before I set up a lot of hardware...

**Reply by s-hadinger** [reply]
200 devices looks like a lot. You may prefer to divide in 3-4 parallel networks with 3-4 coordinators. It will reduce the blast radius if you need to re-pair all devices.

Honestly nobody went to 200 devices on Tasmota yet.

**Reply by DerPicknicker** [reply]
@s-hadinger .. Good idea. What is a possible reason to re-pair a device? This is really a downside about zigbee in comparison to other ip based protocols.

**Reply by sfromis** [reply]
One possible reason to re-pair would be hardware damage to a coordinator device, or theft. Or if you want a new one instead, like having CC2652 for increased capacity.

**Reply by s-hadinger** [reply]
When upgrading zigbee software on the MCU (not Tasmota, the Cc2652p) you may need to re-pair the network

**Reply by DerPicknicker** [reply]
Thank you all.. Is there an api for the Zigbee functions also available for berry Scripting?

**Reply by s-hadinger** [reply]
Yes, you get the regular rules, but there is also a lower level integration
